### DISCUSSION TITLE
API available?

### DISCUSSION THREAD

**Discussion by jzakotnik** [koenidv/sph-planner#46] [General]
hi, I (my kid) is new to Lanis and I was wondering if there's an API, e.g. to include it into Home Assistant Dashboards.. or is the only way screen scraping?

**Comment by spezian** [comment]
There is no official lanis api, so you need to scrape the website to include it on a Home Assistant Dashboard and as a note, I think this project is dead, there is no active maintainer anymore.

**Comment by jzakotnik** [comment]
Thanks for the hint! Sad that no API is there..

**Comment by koenidv** [comment]
Yep, would've been cool if SPH provided an API. Maybe the [Networking package](https://github.com/koenidv/sph-planner/tree/main/app/src/main/java/de/koenidv/sph/networking) can help you create your own interface. But be warned, it was always kind of a headache to work with this website 😅 
If you want to scrape grades and messages you will have to decrypt their "second level encryption" - I solved this by having their JS script run inside a VM. Check this out: https://github.com/koenidv/sph-planner/blob/main/app/src/main/java/de/koenidv/sph/networking/Cryption.kt and #15

Also, yes, this project is dead. I don't have access to SPH anymore and have no interest in continuing this project myself. This is also the reason this app is no longer available on Google Play: They require working credentials to verify the app.

However, if you want to play around with this, go for it! HMU if added or fixed something and can help me get the app published again 🎉

**Comment by aws1313** [comment]
In case you are developing with Dart or Flutter you can use https://gist.github.com/suehok/dfc4a6989537e4a3ba4058669289737f to decrypt their "second level encryption" - has better performance than a VM :)

**Comment by alessioc42** [comment]
It may be a bit late, but there are now a couple of projects that make this possible.
[LanisAPI](https://github.com/kurwjan/LanisAPI) by @kurwjan (Python)
[SPHClient](https://github.com/alessioC42/SPHclient) by me (JavaScript) (but I would also use LanisAPI)
Or use the reverse engineering approach in the app [lanis-mobile](https://github.com/alessioC42/lanis-mobile) (also from @kurwjan  and me) as described in the SPHClient readme if you want more parsing capabilities. (Especially for "mein Unterricht")
